File: src/ClaudeToolbar.Core/Usage/UsageSnapshot.cs
namespace ClaudeToolbar.Core.Usage;

public sealed record UsageSnapshot(
    UsageWindow? FiveHour,
    UsageWindow? SevenDay,
    UsageWindow? SevenDayOpus,
    UsageWindow? SevenDaySonnet,
    DateTimeOffset FetchedAt)
{
    public IEnumerable<UsageWindow> Windows
    {
        get
        {
            if (FiveHour is not null) yield return FiveHour;
            if (SevenDay is not null) yield return SevenDay;
            if (SevenDayOpus is not null) yield return SevenDayOpus;
            if (SevenDaySonnet is not null) yield return SevenDaySonnet;
        }
    }

    public DateTimeOffset? NextReset
    {
        get
        {
            var resets = Windows.Where(w => w.ResetsAt is not null).Select(w => w.ResetsAt!.Value).ToList();
            return resets.Count == 0 ? null : resets.Min();
        }
    }
}

File: mac/Sources/ClaudeToolbarCore/Usage/UsageSnapshot.swift
import Foundation

public struct UsageSnapshot: Equatable, Sendable {
    public let fiveHour: UsageWindow?
    public let sevenDay: UsageWindow?
    public let sevenDayOpus: UsageWindow?
    public let sevenDaySonnet: UsageWindow?
    public let sevenDayFable: UsageWindow?
    public let fetchedAt: Date

    public init(fiveHour: UsageWindow?, sevenDay: UsageWindow?, sevenDayOpus: UsageWindow?, sevenDaySonnet: UsageWindow?, sevenDayFable: UsageWindow? = nil, fetchedAt: Date) {
        self.fiveHour = fiveHour
        self.sevenDay = sevenDay
        self.sevenDayOpus = sevenDayOpus
        self.sevenDaySonnet = sevenDaySonnet
        self.sevenDayFable = sevenDayFable
        self.fetchedAt = fetchedAt
    }

    public var windows: [UsageWindow] {
        [fiveHour, sevenDay, sevenDayOpus, sevenDaySonnet, sevenDayFable].compactMap { $0 }
    }

    public var nextReset: Date? {
        windows.compactMap(\.resetsAt).min()
    }
}
